bftengine/tests/bcstatetransfer/source_selector_test.cpp

```diff
@@ -227,6 +227,48 @@ TEST_F(SourceSelectorTestFixture, select_amongst_preferred_replicas) {
   ASSERT_TRUE(source_selector.isPreferred(source_selector.currentReplica()));
 }
 
+TEST_F(SourceSelectorTestFixture, unknown_primary) {
+  auto source_selector =
+      SourceSelector(replicas, kRetransmissionTimeoutMs, kReplicaReplacementTimeoutMs, maxFetchRetransmissions, GL);
+  source_selector.updateSource(kSampleCurrentTimeMs);  // would set up all preferred replicas internally.
+  ASSERT_TRUE(source_selector.isPreferred(source_selector.currentReplica()));
+  for (uint16_t i = 1; i <= 9; i++) {
```

Comment:
Try to use the macro or config param MAX_PREPREPARE_MSGS.
If you don't have a way to extract it, please do not use hard coded numbers. put above a single constant MAX_PREPREPARE_MSGS with the same name and use it, its a better practice. as a last option you can also consider static assert to make sure they are equal.
this comment is valid for all new tests in file.

Reply:
Thanks. Addressed.